### PR TITLE
Fix type definition of shadow$ and shadow$$

### PR DESCRIPTION
**Pull request by tadashi0713**
## Proposed changes

Fixing this issue about type definition of `shadow$` and `shadow$$`
https://github.com/webdriverio/webdriverio/issues/3911

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tadashi0713** [comment]
@mgrybyk @CrispusDH @christian-bromann Addressed your comments, could you review again if you have time? 🙇

**Comment by CrispusDH** [comment]
I don't see changes for this one https://github.com/webdriverio/webdriverio/pull/3913#pullrequestreview-232947716

**Comment by tadashi0713** [comment]
Opps, will add

**Comment by mgrybyk** [comment]
@tadashi0713  may you please also add `await browser.shadow$('')` to test, please?

**Comment by mgrybyk** [comment]
Actually if we want to have shadow$ in browser context available we need to add it Browser interface as well, otherwise sync tests will fail

**Comment by mgrybyk** [comment]
Ooops, guys, hold on!

**Comment by mgrybyk** [comment]
I missed that `shadow$` commands exist only on Element level, and doesn't exist on browser level!

We don't have to add typings for `shadow$` commands to browser level, why are we doing this?

**Comment by tadashi0713** [comment]
https://github.com/webdriverio/webdriverio/pull/3913#pullrequestreview-232947716
It means we don`t need to do this?

**Comment by mgrybyk** [comment]
@tadashi0713 good question, I don't know to be honest.

Do we need to do something like `shadow$('foo')` or using shadow$ command only possible after getting root element?

**Comment by tadashi0713** [comment]
I thought `shadow$` and `shadow$$` are only possible after getting element
https://webdriver.io/docs/api/element/shadow$.html

**Comment by mgrybyk** [comment]
yeah, looks like you are right, sorry for confusing, @tadashi0713

**Comment by mgrybyk** [comment]
give me a minute, I'll review PR accordingly

File: packages/wdio-sync/webdriverio.d.ts [hunk 1]
```diff
@@ -16,6 +16,8 @@ declare namespace WebdriverIO {
 declare var browser: BrowserObject;
 declare function $(selector: string | Function): WebdriverIO.Element;
 declare function $$(selector: string | Function): WebdriverIO.Element[];
+declare function shadow$(selector: string | Function): WebdriverIO.Element;
```
Comment (mgrybyk) [inline packages/wdio-sync/webdriverio.d.ts]: please remove shadow$ from here

File: packages/webdriverio/webdriverio.d.ts [hunk 2]
```diff
@@ -5,12 +5,14 @@ type $ = (selector: string | Function) => Promise<WebdriverIOAsync.Element>;
...
 
 // Element commands that should be wrapper with Promise
-type ElementPromise = Omit<WebdriverIO.Element, 'addCommand' | '$' | '$$'>;
+type ElementPromise = Omit<WebdriverIO.Element, 'addCommand' | '$' | '$$' | 'shadow$' | 'shadow$$' >;
 
 // Methods which return async element(s) so non-async equivalents cannot just be promise-wrapped
 interface AsyncSelectors {
     $: $;
     $$: $$;
+    shadow$: $;
```
Comment (mgrybyk) [inline packages/webdriverio/webdriverio.d.ts]: please remove shadow$ from here

File: packages/webdriverio/webdriverio.d.ts [hunk 3]
```diff
@@ -49,6 +51,8 @@ declare namespace WebdriverIOAsync {
 declare var browser: BrowserObject;
 declare var $: $;
 declare var $$: $$;
+declare var shadow$: $;
```
Comment (mgrybyk) [inline packages/webdriverio/webdriverio.d.ts]: please remove shadow$ from here

File: packages/webdriverio/webdriverio.d.ts [hunk 4]
```diff
@@ -5,12 +5,14 @@ type $ = (selector: string | Function) => Promise<WebdriverIOAsync.Element>;
 type $$ = (selector: string | Function) => Promise<WebdriverIOAsync.Element[]>;
 
 // Element commands that should be wrapper with Promise
-type ElementPromise = Omit<WebdriverIO.Element, 'addCommand' | '$' | '$$'>;
+type ElementPromise = Omit<WebdriverIO.Element, 'addCommand' | '$' | '$$' | 'shadow$' | 'shadow$$' >;
```
Comment (mgrybyk) [inline packages/webdriverio/webdriverio.d.ts]: not sure if we need this here, checking

Comment (mgrybyk) [inline packages/webdriverio/webdriverio.d.ts]: please remove shadow$* from here

**Comment by mgrybyk** [comment]
@tadashi0713 done. Sorry for confusing you, I supposed that shadow$* commands are available on Browser level as well

File: packages/webdriverio/webdriverio.d.ts [hunk 5]
```diff
@@ -5,7 +5,7 @@ type $ = (selector: string | Function) => Promise<WebdriverIOAsync.Element>;
 type $$ = (selector: string | Function) => Promise<WebdriverIOAsync.Element[]>;
 
 // Element commands that should be wrapper with Promise
-type ElementPromise = Omit<WebdriverIO.Element, 'addCommand' | '$' | '$$'>;
+type ElementPromise = Omit<WebdriverIO.Element, 'addCommand' | '$' | '$$' >;
```
Comment (mgrybyk) [inline packages/webdriverio/webdriverio.d.ts]: redundant space

**Comment by mgrybyk** [comment]
👍

**Comment by tadashi0713** [comment]
@mgrybyk No problem, thanks for review!

**Comment by CrispusDH** [comment]
Sorry for confusing, I was thinking that these commands can be on the same level as `$` and `$$`

**Pull request merged**